src/Popover.js

```diff
@@ -57,40 +54,47 @@ const defaultProps = {
   placement: 'right',
 };
 
-function Popover({
-  bsPrefix,
-  innerRef,
-  placement,
-  className,
-  style,
-  children,
-  content,
-  arrowProps,
-  scheduleUpdate: _,
-  outOfBoundaries: _1,
-  ...props
-}) {
-  return (
-    <div
-      role="tooltip"
-      ref={innerRef}
-      style={style}
-      x-placement={placement}
-      className={classNames(className, bsPrefix, `bs-popover-${placement}`)}
-      {...props}
-    >
-      <div className="arrow" {...arrowProps} />
-      {content ? <PopoverContent>{children}</PopoverContent> : children}
-    </div>
-  );
-}
+const Popover = React.forwardRef(
+  (
+    {
+      bsPrefix,
+      placement,
+      className,
+      style,
+      children,
+      content,
+      arrowProps,
+      scheduleUpdate: _,
+      outOfBoundaries: _1,
```

Comment:
should these just be removed? or are we destructuring them for backwards-compatibility?

Reply:
These come from the react-overlays `<Overlay>`. We're removing them to avoid passing unknown props to DOM children I think.